float32 power
float32 trigger
